sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/LineCountSensor.java

```diff
@@ -44,34 +48,41 @@
   private static final Logger LOG = LoggerFactory.getLogger(LineCountSensor.class);
   private final FileSystem fileSystem;
   private final FilePredicate mainFilesPredicate;
+  private final FileLinesContextFactory fileLinesContextFactory;
 
-  public LineCountSensor(FileSystem fileSystem) {
+  public LineCountSensor(FileSystem fileSystem, FileLinesContextFactory fileLinesContextFactory) {
     this.fileSystem = fileSystem;
     this.mainFilesPredicate = fileSystem.predicates().and(
       fileSystem.predicates().hasType(InputFile.Type.MAIN),
       fileSystem.predicates().hasLanguage(Xml.KEY));
+    this.fileLinesContextFactory = fileLinesContextFactory;
   }
 
-  private static void addMeasures(SensorContext sensorContext, InputFile inputFile, Charset encoding) {
+  private void addMeasures(SensorContext sensorContext, InputFile inputFile, Charset encoding) {
 
     int numLines = 0;
     int numBlankLines = 0;
+    Set<Integer> blankLines = new HashSet<>();
     String lineSeparatorRegexp = "(?:\r)?\n|\r";
 
     try {
       String fileContent = FileUtils.readFileToString(inputFile.file(), encoding.name());
 
       // Count the last empty line
-      if (fileContent.endsWith("\r\n") || fileContent.endsWith("\n")) {
-        numLines++;
-        numBlankLines++;
-      }
+//      if (fileContent.endsWith("\r\n") || fileContent.endsWith("\n")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)